static/app/actionCreators/modal.tsx

```diff
@@ -332,7 +333,13 @@ export async function openWidgetViewerModal({
   onClose,
   ...options
 }: WidgetViewerModalOptions & {onClose?: () => void}) {
-  const mod = await import('sentry/components/modals/widgetViewerModal');
+  const getModal = () => {
+    if (options.widget.widgetType === WidgetType.METRICS) {
+      return import('sentry/components/modals/metricWidgetViewerModal');
+    }
+    return import('sentry/components/modals/widgetViewerModal');
+  };
+  const mod = await getModal();
```

Comment:
No hard opinion here, just my what I would do 😄 
```suggestion
  const modalPromise = options.widget.widgetType === WidgetType.METRICS
    ? import('sentry/components/modals/metricWidgetViewerModal')
    : import('sentry/components/modals/widgetViewerModal');
  
  const mod = await modalPromise;
```

Reply:
Yep, its a bit more compact :D